### ISSUE TITLE
(1.12) Memory Leak: ModConfig#keepEntityAsVanillaCache

### ISSUE THREAD

**Issue by Meldexun** [iwoplaza/MoBends#258]
**Describe the bug**
Entries of the map `ModConfig#keepEntityAsVanillaCache` use entity instances as their keys. This means for every sheep entity for example there will be an entry in the map. Because the map is never cleared this means in theory the map will fill up more and more and require more and more memory.
The other problem with using entities as keys is that the entity object has a world field and thus a lot of other objects will also remain in memory despite not being used anymore.

Changing the map to `Map<Class<? extends Entity>, Boolean>` should fix these issues while still working the same as before.

https://github.com/mobends/MoBends/blob/3b56b02e59fcf2d8571d3fedb6ceee8f8824ba05/src/main/java/goblinbob/mobends/standard/main/ModConfig.java#L40

**To Reproduce**
1. Join a world
2. Leave the world
3. See WorldClient instance not being garbage collected

**Context (please complete the following information):**
 - Minecraft version: 1.12.2
 - Forge version: 14.23.5.2855
 - Mo' Bends version: 1.1.0

**Comment by iwoplaza** [comment]
Good point! An overlook on my part, I'll tend to all the issues you've posted soon. Thanks for reporting them!
